src/components/FlowRunMenu.vue

```diff
@@ -1,7 +1,7 @@
 <template>
   <p-icon-button-menu>
     <template #default>
-      <!-- todo: this PR(https://github.com/PrefectHQ/prefect-ui-library/pull/1743) will only partially work once object acls are on -->
+      <!-- todo: audit [cchoy] this PR(https://github.com/PrefectHQ/prefect-ui-library/pull/1743) will only partially work once object acls are on -->
```

Comment:
For context, looking at what this PR solves is a good example of what I'm trying to fix now. So this past PR added this `can.run.deployment` check so that the button won't render if the current user cannot actually run deployments.

However, now that object ACLs are around and they can grant _additional_ permissions to an actor beyond what they may otherwise have at the workspace level, this check could now evaluate to false when the actor actually _can_ run the specific deployment that's in context because they're explicitly given run access via the object acl but don't have `can.run.deployment`

Reply:
For this I'd would just get the deployment and check it. We can get individual deployments and let the subscriptions and the request batcher take care of caching and optimizations.